Mathlib/Order/Filter/CardinalInter.lean

```diff
@@ -152,4 +156,182 @@ theorem EventuallyEq.cardinal_bInter {S : Set ι} (hS : #S < c)
   (EventuallyLE.cardinal_bInter hS fun i hi => (h i hi).le).antisymm
     (EventuallyLE.cardinal_bInter hS fun i hi => (h i hi).symm.le)
 
+/-- Construct a filter with cardinal `c` intersection property. This constructor deduces
+`Filter.univ_sets` and `Filter.inter_sets` from the cardinal `c` intersection property. -/
+def ofCardinalInter (l : Set (Set α)) (hc : 2 < c)
+    (hp : ∀ S : Set (Set α), (#S < c) → S ⊆ l → ⋂₀ S ∈ l)
+    (h_mono : ∀ s t, s ∈ l → s ⊆ t → t ∈ l) : Filter α where
+  sets := l
+  univ_sets :=
+    sInter_empty ▸ hp ∅ (mk_eq_zero (∅ : Set (Set α)) ▸ lt_trans zero_lt_two hc) (empty_subset _)
+  sets_of_superset := h_mono _ _
+  inter_sets {s t} hs ht := sInter_pair s t ▸ by
+    apply hp _ (?_) (insert_subset_iff.2 ⟨hs, singleton_subset_iff.2 ht⟩)
+    have : #({s, t} : Set (Set α)) ≤ 2 := by
+      calc
+      _ ≤ #({t} : Set (Set α)) + 1 := Cardinal.mk_insert_le
+      _ = 2 := by norm_num
+    exact lt_of_le_of_lt this hc
+
+lemma cardinalInter_ofCardinalInter (l : Set (Set α)) (hc : 2 < c)
+    (hp : ∀ S : Set (Set α), (#S < c) → S ⊆ l → ⋂₀ S ∈ l)
+    (h_mono : ∀ s t, s ∈ l → s ⊆ t → t ∈ l) :
+    CardinalInterFilter (Filter.ofCardinalInter l hc hp h_mono) c :=
+  ⟨hp⟩
+
+@[simp]
+theorem mem_ofCardinalInter {l : Set (Set α)} (hc : 2 < c)
+    (hp : ∀ S : Set (Set α), (#S < c) → S ⊆ l → ⋂₀ S ∈ l) (h_mono : ∀ s t, s ∈ l → s ⊆ t → t ∈ l)
+    {s : Set α} : s ∈ Filter.ofCardinalInter l hc hp h_mono ↔ s ∈ l :=
+  Iff.rfl
+
+/-- Construct a filter with cardinal `c` intersection property.
+Similarly to `Filter.comk`, a set belongs to this filter if its complement satisfies the property.
+Similarly to `Filter.ofCardinalInter`,
+this constructor deduces some properties from the cardinal `c` intersection property
+which becomes the cardinal `c` union property because we take complements of all sets.
+
+Another small difference from `Filter.ofCardinalInter`
+is that this definition takes `p : Set α → Prop` instead of `Set (Set α)`. -/
+def ofCardinalUnion (p : Set α → Prop) (hc : 2 < c)
+    (hUnion : ∀ S : Set (Set α), (#S < c) → (∀ s ∈ S, p s) → p (⋃₀ S))
+    (hmono : ∀ t, p t → ∀ s ⊆ t, p s) : Filter α := by
+  refine .ofCardinalInter {s | p sᶜ} hc (fun S hSc hSp ↦ ?_) fun s t ht hsub ↦ ?_
```

Comment:
Is there any reason for this "small difference"? As in, could we have:
```suggestion
which becomes the cardinal `c` union property because we take complements of all sets. -/
def ofCardinalUnion (l : Set (Set α)) (hc : 2 < c)
    (hUnion : ∀ S : Set (Set α), (#S < c) → (∀ s ∈ S, s ∈ l) → ⋃₀ S ∈ l)
    (hmono : ∀ t ∈ l, ∀ s ⊆ t, s ∈ l) : Filter α := by
  refine .ofCardinalInter {s | sᶜ ∈ l} hc (fun S hSc hSp ↦ ?_) fun s t ht hsub ↦ ?_
```
If so please make the corresponding change for `CountableInterFilter`.

Reply:
(and rename `p` to `l` below)

Reply:
I just adapted the corresponding ones from `CountableInterFilter`, I'll check later today!

Reply:
@ocfnash can you check if you like my proof in the last two PR's? If so, I'll merge!